app/screens/more_channels/more_channels.js

```diff
@@ -433,12 +471,22 @@ export default class MoreChannels extends PureComponent {
 
             let activeChannels = channels;
 
-            if (canShowArchivedChannels && typeOfChannels === 'archived') {
+            if (typeOfChannels === 'shared') {
+                activeChannels = sharedChannels;
+            } else if (canShowArchivedChannels && typeOfChannels === 'archived') {
                 activeChannels = archivedChannels;
             }
 
             let channelDropdown;
             if (canShowArchivedChannels) {
+                let channelDropdownText;
+                if (typeOfChannels === 'public') {
+                    channelDropdownText = formatMessage({id: 'more_channels.showPublicChannels', defaultMessage: 'Show: Public Channels'});
+                } else if (typeOfChannels === 'shared') {
+                    channelDropdownText = formatMessage({id: 'more_channels.showSharedChannels', defaultMessage: 'Show: Shared Channels'});
+                } else {
+                    channelDropdownText = formatMessage({id: 'more_channels.showArchivedChannels', defaultMessage: 'Show: Archived Channels'});
+                }
```

Comment:
this dropdown text is only being set if `canShowArchivedChannels` is set to true